src/containers/ResubmitSolutionContainer/ResubmitSolutionContainer.js

```diff
@@ -15,32 +16,51 @@ import {
 import EvaluationProgressContainer from '../EvaluationProgressContainer';
 import withLinks from '../../helpers/withLinks';
 import { fetchSubmissionEvaluationsForSolution } from '../../redux/modules/submissionEvaluations';
+import { getProgressObserverId } from '../../redux/selectors/evaluationProgress';
 
-const ResubmitSolutionContainer = ({
-  id,
-  assignmentId,
-  userId,
-  resubmit,
-  monitor,
-  isProcessing,
-  newSubmissionId,
-  fetchSubmissions,
-  isDebug = true,
-  links: { SUBMISSION_DETAIL_URI_FACTORY }
-}) => {
-  return (
-    <span>
-      <ResubmitSolution id={id} resubmit={resubmit} isDebug={isDebug} />
-      <EvaluationProgressContainer
-        isOpen={isProcessing}
-        monitor={monitor}
-        link={SUBMISSION_DETAIL_URI_FACTORY(assignmentId, newSubmissionId)}
-        onFinish={() => fetchSubmissions(userId)}
-        onUserClose={() => fetchSubmissions(userId)}
-      />
-    </span>
-  );
-};
+const getResubmitObserverId = defaultMemoize(
```

Comment:
A little, yes. But since this is only a trivial function and since both containers should be independent, I decided to keep it separately in both files.